File: Intermediate/Problem_11.cpp
//Insertion sort non-recursive approach

#include<iostream>
using namespace std;

void insertion_sort(int *arr, int n)
{
    int i,j,temp;
    for(i = 1; i < n; i++)
    {
        temp = arr[i];
        j = i - 1;
        while(j >= 0)
        {
            if(arr[j] > temp)
            {
                arr[j+1] = arr[j];
            }
            else
            {
                break;
            }
            j--;
        }
        arr[j+1] = temp;
    }
}

int main()
{
    int n;
    cin>>n;
    int *arr = new int[n];
    for(int i = 0; i < n; i++)
    {
        cin>>arr[i];
    }
    insertion_sort(arr, n);
    for(int i = 0; i < n; i++)
    {
        cout<<arr[i]<<" ";
    }
    return 0;

}

File: Intermediate/Java/problem_11.java
/*Write a program for insertion sort*/
//Solved by INikhita
import java.util.Arrays;
class InsertionSort {
  void insertionSort(int array[]) {
    int size = array.length;
    for(int step = 1;step<size;step++) {
      int key = array[step];
      int j = step - 1;
      while(j >= 0 && key < array[j]) {
        array[j + 1] = array[j];
         --j;
      }
      array[j + 1] = key;
    }
  }
  public static void main(String args[]) {
    int[] data = { 9, 5, 1, 4, 3 };
    InsertionSort is = new InsertionSort();
    is.insertionSort(data);
    System.out.println("Sorted Array in Ascending Order: ");
    System.out.println(Arrays.toString(data));
  }
}
